### PR TITLE
Tango: style Quick Effect selector

### PR DESCRIPTION
**Pull request by ronso0**
based on #4501

**updated:**
image

### CODE REVIEW

**Review by daschuer — CHANGES_REQUESTED** [review]
Please rebase this to main. Without a rebase we see rebased commits from the already merged PRs twice in the history.

**Comment by daschuer** [comment]
The rest looks good.

**Comment by ronso0** [comment]
> Please rebase this to main. Without a rebase we see rebased commits from the already merged PRs twice in the history.

That's how I planned to do it.
I wonder why the log doesn't have duplicate commits after merging the LateNight fx PR.

**Review by Be-ing — CHANGES_REQUESTED** [review]
The removal of the border and making the background identical to the surroundings makes it hard to tell that the comboboxes are comboboxes. This PR is a step backwards IMO. Can you explain why you think these changes are an improvement?

**Comment by daschuer** [comment]
> I wonder why the log doesn't have duplicate commits after merging the LateNight fx PR.

These commits where identical, inclusive their history. Here they are almost identical, with a different commit hash.

**Review by daschuer — APPROVED** [review]

File: res/skins/Tango/style.qss
Comment (Be-ing) [inline res/skins/Tango/style.qss:1737]: Please revert this.

**Comment by Swiftb0y** [comment]
IMO its still visible that the effect name is a clickable combobox because of the "down-arrowhead" next to the name. 
Removing the border to remove visual clutter makes sense here.

**Comment by Holzhaus** [comment]
> The removal of the border and making the background identical to the surroundings makes it hard to tell that the comboboxes are comboboxes. This PR is a step backwards IMO. Can you explain why you think these changes are an improvement?

I think it mainly saves screen space that would otherwise be used by margins/pasding/borders. I'm also torn on this, because I think it space-efficient and looks good, but on the other hand it's less easy to discover. Maybe we can consider adding borders and reducing font size instead.

**Comment by Swiftb0y** [comment]
If we talk about discoverability, I think thats fine too. It's not an essential feature that you need to see understand and instantly (like the playbutton for example). Its not like its deeply buried in menus (unless we hide it using a skin or preference option, which is not good idea IMO).

**Comment by Be-ing** [comment]
I agree that it isn't crucial that this is super easy to discover and even with this change it isn't too hard to discover, but still, this makes it harder to discover and I don't think there was a problem with the styling before.

**Comment by Swiftb0y** [comment]
I guess its a personal taste call, and I prefer this. Should we decide via a quick poll?

**Comment by Be-ing** [comment]
I would like to hear @ronso0's motivation for this change before jumping to a vote.

**Comment by ronso0** [comment]
My motivation for this plain effect selector
* with borders it's the inner padding + border + margin to the EQs and bottom container border would make the entire mixer 'much' taller and the decks would grow unnecessarily -- especially in Tango which is meant to be as compact as possible
* it has the same drop-down arrow indicating a combobox as the effect selector, the AutoDJ mode selector and the search box, as well as a hover effect, thus I consider it discoverable
* I do not expect the majority of users to change this frequently, if at all, thus styling it more like an informative label makes much more sense than adding the visual clutter of a fully decorated combobox

**Review by Be-ing — CHANGES_REQUESTED** [review]
There is less room for text with this branch. Here is main:
image

This branch:
image

**Comment by ronso0** [comment]
The previous implementation was quick and dirty and always added extra row to the deck.
It was okay for testing but I'm not going to premanently change the layout that much just to have a large quick effect selector.

Though, I can try to squeeze out a few more pixels for the selector.

**Comment by ronso0** [comment]
image

Note: the screenshot was made with #4532

**Comment by ronso0** [comment]
marked as draft because it looks like I broke the fx units. I'll remove the extra tweaks and add the chain preset button styling asap.

**Comment by Be-ing** [comment]
I wouldn't be opposed to that last screenshot, though I still prefer how it is currently in the main branch.

**Comment by ronso0** [comment]
image

Ready.

**Comment by ronso0** [comment]
In another PR I'll make the fx unit menu more discoverable (WEffectChainPresetButton), the arrow was just a quick shot, too. I used the gear icon (skin settings) for that in LateNight and it makes more sense than the down arrow.

**Comment by Swiftb0y** [comment]
Why did you put the PFL button the same height as the dropdown? Wouldn't it make more sense to make the vumeter shorter to put the pfl button horizontally aligned with the filter and gain knob so there is more space for the text in dropdown?

**Comment by ronso0** [comment]
Yes, that would look better for that one configuration.
But it has some consequences and requires more elaborate changes to keep the skin consistent (trying to keep my initial design intact).
* channel VUs are aligned with the main VU (shrink the main VU, too?)
* with EQ knobs hidden the Vol slider and VU should expand to available height. The VU pixmap are designed pixel-perfect so we'd need two VU meters per deck? Or have one version center vertically?

Tbh I'm not motivated to work on that right now. The Deere implementation is still overdue, LateNight still needs the chain menu button...

**Comment by ronso0** [comment]
That said, I'd like to merge this now and me or anyone else may propose improvements later on.

**Comment by Swiftb0y** [comment]
Thanks for the explanation. Sure we can merge.

**Comment by ronso0** [comment]
@Be-ing already agreed to the new implementationearlier so I don't think we need to wait for his formal approval.

**Comment by Swiftb0y** [comment]
He also recently announced to abstain from contributing to mixxx for the foreseeable future, I don't think he'll care to review/approve this PR because of that.

**Comment by ronso0** [comment]
_pa-ping..._

**Comment by Holzhaus** [comment]
Alright, shall we merge then?

**Review by Swiftb0y — APPROVED** [review]
Thanks :+1:

**Pull request merged**
